daemon/cmd/kube_proxy_healthz_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package cmd
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/cilium/cilium/api/v1/models"
+
+	. "gopkg.in/check.v1"
+)
+
+// 'check' testing suite scaffolding.
+type KubeProxyHealthzTestSuite struct{}
+
+var _ = Suite(&KubeProxyHealthzTestSuite{})
+
+// Injected fake service.
+type FakeService struct {
+	injectedCurrentTs     time.Time
+	injectedLastUpdatedTs time.Time
+}
+
+func (s *FakeService) GetCurrentTs() time.Time {
+	return s.injectedCurrentTs
+}
+
+func (s *FakeService) GetLastUpdatedTs() time.Time {
+	return s.injectedLastUpdatedTs
+}
+
+// Injected fake daemon.
+type FakeDaemon struct {
+	injectedStatusResponse models.StatusResponse
+}
+
+func (d *FakeDaemon) getStatus(blah bool) models.StatusResponse {
+	return d.injectedStatusResponse
+}
+
+type healthzPayload struct {
+	LastUpdated string
+	CurrentTime string
+}
+
+func (s *KubeProxyHealthzTestSuite) TestKubeProxyHealth(c *C) {
+	lastUpdateTs := time.Unix(100, 0) // Fake 100 seconds after Unix.
+	currentTs := time.Unix(200, 0)    // Fake 200 seconds after Unix.
+	expectedTs := currentTs
```

Comment:
nit: you can move `lastUpdateTs`, `currentTs` and `expectedTs` to the `healthTestHelper` function, so you will not have to pass them as arguments ;)

Reply:
sure will check this out.